corgidrp/data.py

```diff
@@ -724,6 +756,177 @@ def copy(self, copy_data = True):
+        if ext_hdr is not None:
+            if input_dataset is None and 'DRPNFILE' not in ext_hdr.keys():
+                # error check. this is required in this case
+                raise ValueError("This appears to be a new noise map. The dataset of input files needs to be passed in to the input_dataset keyword to record history of this noise map.")
+
+            self.ext_hdr['DATATYPE'] = 'NoiseMap' # corgidrp specific keyword for saving to disk
+            self.ext_hdr['BUNIT'] = 'detected electrons'
+
+            # log all the data that went into making this calibration file
+            if 'DRPNFILE' not in ext_hdr.keys():
+                self._record_parent_filenames(input_dataset)
+            # add to history
+            self.ext_hdr['HISTORY'] = noise_type + " NoiseMap calibration file created"
+
+            # give it a default filename
+            orig_input_filename = self.ext_hdr['FILE0'].split(".fits")[0]
+            self.filename = "{0}_{1}_NoiseMap.fits".format(orig_input_filename, noise_type)
+
+        if err_hdr is not None:
+            self.err_hdr['BUNIT'] = 'detected electrons'
+
+
+        # double check that this is actually a KGain file that got read in
+        # since if only a filepath was passed in, any file could have been read in
+        if 'DATATYPE' not in self.ext_hdr or self.ext_hdr['DATATYPE'] != 'NoiseMap':
+            raise ValueError("File that was loaded was not a " + noise_type + " NoiseMap Calibration file.")
+
+
+    def copy(self, copy_data = True):
+        """
+        Make a copy of this NoiseMap file, including data and headers.
+        Data copying can be turned off if you only want to modify the headers
+        Headers should always be copied as we should modify them any time we make new edits to the data
+
+        Args:
+            copy_data (bool): (optional) whether the data should be copied. Default is True
+
+        Returns:
+            new_nm (corgidrp.data.NoiseMap): a copy of this NoiseMap
+        """
+        if copy_data:
+            new_data = np.copy(self.data)
+            new_err = np.copy(self.err)
+            new_dq = np.copy(self.dq)
+        else:
+            new_data = self.data # this is just pointer referencing
+            new_err = self.err
+            new_dq = self.dq
+        new_nm = NoiseMap(new_data, noise_type=self.noise_type, pri_hdr=self.pri_hdr.copy(), ext_hdr=self.ext_hdr.copy(), err = new_err, dq = new_dq, err_hdr = self.err_hdr.copy())
+
+        # annoying, but we got to manually update some parameters. Need to keep track of which ones to update
+        new_nm.filename = self.filename
+        new_nm.filedir = self.filedir
+
+        # update DRP version tracking
+        self.ext_hdr['DRPVERSN'] =  corgidrp.version
+        self.ext_hdr['DRPCTIME'] =  time.Time.now().isot
+
+        return new_nm
+
+class BiasOffset(Image):
```

Comment:
Is it used in other steps or calibrations?

Reply:
This class exists because bias offset is a calibration product that comes out of the calibrate_darks_lsq() function.  That value would then be an input for the prescan bias subtraction step function (or I guess that function would have as an input an instance of the BiasOffset class).

Reply:
Can we just integrate it as part of the NoiseMaps class?

Reply:
Ahhh, I see. This would be the input to the prescan_bias that is currently a `bias_offset` keyword? 

`def prescan_biassub(input_dataset, bias_offset=0., return_full_frame=False):`